src/pages/index.js

```diff
@@ -0,0 +1,348 @@
+
+        if (isLiked) {
+            api.removeLike(cardId)
+                .then((updatedCard) => {
+                    cardLikeBtn.classList.remove("card__like-button_liked");
+                    // Optionally update like counter here
+                })
+                .catch(console.error);
+        } else {
+            api.addLike(cardId)
+                .then((updatedCard) => {
+                    cardLikeBtn.classList.add("card__like-button_liked");
+                    // Optionally update like counter here
+                })
+                .catch(console.error);
+        }
+    });
+
+    cardImageEl.addEventListener("click", () => {
+        openModal(previewModal);
+        previewModalImageEl.src = data.link;
+        previewModalCaptionEl.textContent = data.name;
+        previewModalImageEl.alt = data.name;
+    });
+
+    cardDeleteBtn.addEventListener("click", () => {
+        handleDeleteCard(cardElement, data._id);
+    });
+
+    return cardElement;
+}
+
+function openModal(modal) {
+    modal.classList.add("modal_opened");
+    document.addEventListener("keyup", handleEscape);
+}
+
+function closeModal(modal) {
+    modal.classList.remove("modal_opened");
+    document.removeEventListener("keyup", handleEscape);
+}
+
+function handleEditFormSubmit(evt) {
+    evt.preventDefault();
+
+    const submitBtn = evt.submitter;
+    setbuttonText(submitBtn, true, "Saving...", "Save");
+
+    api.editUserInfo({
+        name: editModalNameInput.value,
+        about: editModalDescriptionInput.value,
+    })
+        .then((data) => {
+            profileName.textContent = data.name;
+            profileDescription.textContent = data.about;
+            closeModal(editProfileModal);
+        })
+        .catch(console.error)
+        .finally(() => {
+            setbuttonText(submitBtn, false, "Saving...", "Save");
+        });
+}
+
+function handleAddCardSubmit(evt) {
+    evt.preventDefault();
+
+    const inputValues = {
+        name: cardNameInput.value,
+        link: cardLinkInput.value,
+    };
+
+    api.addCard(inputValues)
+        .then((cardData) => {
+            const cardElement = getCardElement(cardData);
+            cardsList.prepend(cardElement);
+            evt.target.reset();
+            disabledButton(cardSubmitBtn, settings);
+            closeModal(cardModal);
+        })
+        .catch(console.error);
+}
+
+function handleAvatarSubmit(evt) {
+    evt.preventDefault();
+
+    const submitBtn = evt.submitter || avatarSubmitBtn;
+    setbuttonText(submitBtn, true, "Saving...", "Save");
+
+    const avatarUrl = avatarInput.value;
+    if (!avatarUrl) {
+        console.error("Avatar URL is empty.");
+        setbuttonText(submitBtn, false, "Saving...", "Save");
+        return;
+    }
+
+    api.editAvatarInfo(avatarUrl)
+        .then((data) => {
+            if (data && data.avatar) {
+                const profileAvatar =
+                    document.querySelector(".profile__avatar");
```

Comment:
`        const profileAvatar = document.querySelector(".profile__avatar");
`

You've already found the avatar in line 65 as `avatarModalBtn`. Please, use it here

Reply:
* Another remark hasn't been corrected https://github.com/Luidelson/se_project_spots/pull/1#discussion_r2094621833